### PR TITLE
add test-mac 

### PR DESCRIPTION
**Pull request by sahiljhawar**
Solves #255 by adding test-mac.yml. For Conda test mamba is not working on MacOS so switched to Conda

### CODE REVIEW

**Comment by sahiljhawar** [comment]
@JohannesBuchner Please review

File: .github/workflows/test-mac.yml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+    - name: Coveralls Finished
+      uses: coverallsapp/github-action@master
+      if: ${{ matrix.MPI == 'no' }}
+      with:
+        path-to-lcov: lcov.info
+        github-token: ${{ secrets.github_token }}
+        flag-name: MPI-${{ matrix.MPI }}
+
+  CondaTest:
```
Comment (JohannesBuchner) [inline .github/workflows/test-mac.yml:121]: please remove this and everything below

**Comment by JohannesBuchner** [comment]
Thank you @sahiljhawar for this PR!

**Comment by JohannesBuchner** [comment]
I thought it was working fine, but now it complains that there is no Fortran compiler installed? https://github.com/JohannesBuchner/PyMultiNest/actions/runs/10465458463

**Comment by sahiljhawar** [comment]
it was indeed working fine. I can look into it later, and see where is gfortran located

**Pull request merged**